torch/fx/experimental/symbolic_shapes.py

```diff
@@ -111,10 +112,47 @@ class PySymInt(object):
     our program. They're what sit under FakeTensor, and contains our primary
     implementation of symbolic shapes.
     """
-    def __init__(self, expr, shape_env, constant=None):
+
+    class TensorReference(object):
```

Comment:
Just put this top level

Reply:
Make this a dataclass?

Reply:
I dislike data classes. I have a harder time putting breakpoints in them, but if you prefer it, I can do that.

Reply:
Okay, I can make it a dataclass